tests/cli/bootprep/test_vars.py

```diff
@@ -0,0 +1,127 @@
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+Unit tests for variable loading
+"""
+
+from textwrap import dedent
+import unittest
+from unittest.mock import mock_open, patch
+
+from sat.cli.bootprep.constants import LATEST_VERSION_VALUE
+from sat.cli.bootprep.vars import VariableContext
+
+
+class TestEnumeratingVars(unittest.TestCase):
+    """Test enumerating variables and sources"""
+
+    def setUp(self):
+        self.context = VariableContext()
+        self.vars_file_contents = dedent("""
+        foo:
+            version: 2.3.4
+        bar:
+            version: 4.5.6
+        """)
+
+        self.mock_catalog = patch('sat.cli.bootprep.vars.HPCSoftwareRecipeCatalog').start()
+        self.software_recipe_vars = {
+            'foo': {'version': '1.2.3'},
+            'bar': {'version': '2.3.4'}
+        }
+        self.mock_catalog.return_value.get_latest_version.return_value.all_vars = self.software_recipe_vars
+
+    def tearDown(self):
+        patch.stopall()
+
+    def test_getting_latest_recipe_vars(self):
+        """Test enumerating variables from the latest recipe"""
+        self.context.recipe_version = LATEST_VERSION_VALUE
+        expected_elements = {('foo.version', '1.2.3', 'latest recipe'),
+                             ('bar.version', '2.3.4', 'latest recipe')}
+        for element in list(self.context.enumerate_vars_and_sources()):
+            self.assertIn(element, expected_elements)
+
+    def test_getting_arbitrary_recipe_vars(self):
+        """Test enumerating variables from a given recipe version"""
+        self.context.recipe_version = '22.07'
+        expected_elements = {('foo.version', '1.2.3', '22.07 recipe'),
+                             ('bar.version', '2.3.4', '22.07 recipe')}
+        for element in list(self.context.enumerate_vars_and_sources()):
+            self.assertIn(element, expected_elements)
```

Comment:
This is returning an empty dict, which means that we're not actually testing whether the expected values are returned with the `self.assertIn` call below. Doing this assertion instead seems to fix the problem.

```suggestion
        self.assertEqual(expected_elements,
                         set(self.context.enumerate_vars_and_sources()))
```

I would do the same in the other test methods.